http/http-advanced-reactive/src/test/java/io/quarkus/ts/http/advanced/reactive/Http2IT.java

```diff
@@ -0,0 +1,259 @@
+                            .compose(req -> req.send()
+                                    .compose(httpClientResponse -> {
+                                        requestCheckpoint.flag();
+                                        assertEquals(HttpStatus.SC_OK, httpClientResponse.statusCode());
+                                        assertEquals(HttpVersion.HTTP_2, httpClientResponse.version());
+                                        return httpClientResponse.body();
+                                    }))
+                            .onSuccess(body -> {
+                                assertThat("Body response", body.toString().contains(GREETING));
+                                requestCheckpoint.flag();
+                            }).onFailure(throwable -> vertxTestContext.failNow(throwable.getMessage()));
+                });
+    }
+
+    @Test
+    @Order(3)
+    @DisplayName("HttpClient Vertx just with HTTP/2 protocol option set and default port")
+    void http2ProtocolTest(Vertx vertx, VertxTestContext vertxTestContext) {
+        HttpClientOptions httpClientOptions = new HttpClientOptions()
+                .setTrustOptions(new JksOptions().setPath("keystore.jks").setPassword("password"))
+                .setProtocolVersion(HttpVersion.HTTP_2);
+        httpClient = vertx.httpClientBuilder().with(httpClientOptions).build();
+        httpClient.request(HttpMethod.GET, baseUri.getPort(), baseUri.getHost(), BASE_ENDPOINT)
+                .compose(request -> request.send()
+                        .compose(httpClientResponse -> {
+                            assertEquals(HttpStatus.SC_OK, httpClientResponse.statusCode());
+                            assertEquals(HttpVersion.HTTP_2, httpClientResponse.version());
+                            return httpClientResponse.body();
+                        }))
+                .onSuccess(body -> vertxTestContext.verify(() -> {
+                    assertThat("Body response", body.toString().contains(GREETING));
+                    vertxTestContext.completeNow();
+                })).onFailure(Throwable::printStackTrace);
+
+    }
+
+    @Test
+    @DisplayName("Test max header size exceeded error message")
+    @Order(5)
+    void testMaxHeaderSizeExceeded(Vertx vertx, VertxTestContext testContext) {
+        HttpClientOptions options = new HttpClientOptions()
+                .setSsl(true)
+                .setUseAlpn(true)
+                .setTrustOptions(new JksOptions().setPath("keystore.jks").setPassword("password"))
+                .setProtocolVersion(HttpVersion.HTTP_2);
+        httpClient = vertx.httpClientBuilder().with(options).build();
+
+        int expectedLimit = 200;
+        int headerCount;
+
+        for (int i = 0; i < expectedLimit; i++) {
+            headerCount = i;
+
+            try {
+                request(headerCount, testContext);
+            } catch (Exception e) {
```

Comment:
```suggestion

```
Please mention one example when there can be an exception. One is enough. I can't imagine that `e.getMessage().contains("Header size exceeded max allowed size")` will be true and it is not a bug in Vert.x. Can you point me to docs that says that? Vertx. should handle failure as standard state and not to throw exceptions.